packages/lib/HtmlToMd.ts

```diff
@@ -26,8 +27,21 @@ export default class HtmlToMd {
 		turndown.use(turndownPluginGfm);
 		turndown.remove('script');
 		turndown.remove('style');
+		const pdfRule = {
+			filter: ['embed', 'object'],
+			replacement: function(_content: string, node: any, _options: any) {
+				if (node.getAttribute('src') && (node.getAttribute('type') === 'application/pdf' || node.getAttribute('src').endsWith('.pdf'))) {
+					// We are adding _pdf: prefix so that we can later distingish them from normal links and create resources for them.
+					return `[${node.getAttribute('src')}](_pdf:${node.getAttribute('src')})`;
+				}
+				return '';
+			},
+		};
+		if (options.loadPdfs) {
+			turndown.addRule('pdf', pdfRule);
+		}
```

Comment:
HtmlToMd is a generic HTML to Markdown converter so it's not ideal to have this _pdf trick here, but I see why you've done that, and I can't think of any simple alternative. Also the fact that it produces invalid Markdown is not great.

I would suggest the following:

- Rename the option to something that makes it clearer it's a special case, such as `convertEmbeddedPdfsToLinks`

- Instead of setting the link content to the source, couldn't you use that to mark the PDFs that need to be imported? eg `[embedded_pdf](${node.getAttribute('src'))`. That way you know that all the "embedded_pdf" links need to be imported, and also the resulting Markdown is now valid. So it will work even if the process to download the PDF files fails.

Reply:
The 2nd point was infact my approach at first but I later found out that when we extract file urls using `markdownIt.parse()` it does not have an api to extract the first part (i.e `embedded_pdf`) of the link and I was left with this option. https://github.com/laurent22/joplin/blob/cc920ed03b62aadff23815801b386a2f5e5b9010/packages/lib/markdownUtils.ts#L78

Also see the api: https://markdown-it.github.io/markdown-it/#MarkdownIt.parse

Reply:
This is the token object I'm receiving for the pdf link, as you can see, the text was omitted. 

```json
{
    "type": "link_open",
    "tag": "a",
    "attrs": [
        [
            "href",
            "_pdf:https://pdfobject.com/pdf/sample-3pp.pdf"
        ]
    ],
    "map": null,
    "nesting": 1,
    "level": 0,
    "children": null,
    "content": "",
    "markup": "",
    "info": "",
    "meta": null,
    "block": false,
    "hidden": false
}
```

On some searching I found out this post https://discourse.joplinapp.org/t/help-with-markdown-it-link-rendering/8143/5

where you commented:
> I’ve done that before and that was a mistake as their parse method really shouldn’t be used directly. Instead use the regular render() method along with a custom plugin you’d write. From the plugin you can access all the tokens, like from parse but in a cleaner way.

Maybe we can try this.

Reply:
Ohh I see, should have checked all the tokens once, my bad. Thanks a lot for the help, this indeed worked out!